crates/omnix-common/src/config.rs

```diff
@@ -24,8 +24,38 @@ pub struct OmConfig {
 }
 
 impl OmConfig {
-    /// Read the om configuration from the flake url
-    pub async fn from_flake_url(cmd: &NixCmd, flake_url: &FlakeUrl) -> Result<Self, OmConfigError> {
+    /// Fetch the `om` configuration from `om.yaml` if present, falling back to `om` config in flake output
+    pub async fn get(cmd: &NixCmd, flake_url: &FlakeUrl) -> Result<Self, OmConfigError> {
+        match Self::from_yaml(cmd, flake_url).await {
+            Err(OmConfigError::YamlNotFound(_)) => Self::from_flake(cmd, flake_url).await,
```

Comment:
I just realize I got confused by `YamlNotFound`. Because it a value of OmConfig**Error**, I somehow confused it to refer to yaml **errors**. 

Just get rid of `YamlNotFound` and have `from_yaml` return `Option<Self>` instead, so the caller do fallback on `None`.

Reply:
If `from_yaml` returns `Option<Self>` will we not be silently skipping over errors while parsing the YAML?

Reply:
No, I meant:

```rust
    async fn from_yaml(cmd: &NixCmd, flake_url: &FlakeUrl) -> Result<Option<Self>, OmConfigError> {
```

Reply:
Done in https://github.com/juspay/omnix/pull/352/commits/5ba02223e4165923a5050cc2d2a9b982cc4252a2